sdks/python/apache_beam/runners/portability/fn_api_runner/visualization_tools.py

```diff
@@ -0,0 +1,115 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Set of utilities to visualize a pipeline to be executed by FnApiRunner."""
+from typing import Set
+from typing import Tuple
+
+from apache_beam.runners.portability.fn_api_runner.translations import Stage
+from apache_beam.runners.portability.fn_api_runner.watermark_manager import WatermarkManager
+
+
+def show_stage(stage: Stage):
+  try:
+    import graphviz
+  except ImportError:
+    import warnings
+    warnings.warn('Unable to draw pipeline. graphviz library missing.')
```

Comment:
That's right - since the utilities are only used for internal debugging of the runner, I hesitate to add it as a dependency - it is also large as you point out. I don't think it fits under the `test` tag dependencies - we may need an extra tag for `internal` dependencies or something like that. I'm happy to add a new tag, or leave it out. Thoughts?